### PR TITLE
fix: Run ts coverage check with test

### PR DESCRIPTION
**Pull request by stocaaro**
#### Description of changes
Enable ts coverage check and reporting with each `npm run test` execution.

This is already enabled for several categories, but this change turns it on for the remaining categories.

#### Description of how you validated changes
Verified that `npm run test` includes a coverage check for each category.

#### Checklist
- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by israx — DISMISSED** [review]
lgtm. Thank you for putting this together and fixing the non running tests.

Maybe it is out of the scope for this PR, but I'm wondering if there is the need to update the current threshold for some packages as TS improvements have been applied.

**Review by israx — APPROVED** [review]

**Review by svidgen — DISMISSED** [review]

File: packages/predictions/tslint.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	"jsRules": {},
 	"rules": {
 		"prefer-const": true,
-		"max-line-length": [true, 120],
+		"max-line-length": false,
```
Comment (cshfang) [inline packages/predictions/tslint.json:8]: Why is this being changed?

Comment (stocaaro) [reply packages/predictions/tslint.json:8]: The linter gets hung up on the comments in this file with max-line-length turned on.
https://github.com/aws-amplify/amplify-js/blob/main/packages/predictions/src/types/AWSTypes.ts

File: packages/pushnotification/package.json [hunk 2]
```diff
@@ -9,14 +9,15 @@
...
-    "test": "tslint 'src/**/*.ts' && jest -w 1 --coverage",
+    "test": "npm run lint && jest -w 1 --coverage",
     "build-with-test": "npm test && npm run build",
     "build": "npm run clean && tsc",
     "build:cjs:watch": "node ./build es5 --watch",
     "build:esm:watch": "echo \"ES Modules are not exported in react-native. Use target build:cjs:watch\"",
     "clean": "rimraf lib lib-esm dist",
     "format": "echo \"Not implemented\"",
-    "lint": "tslint 'src/**/*.ts'"
+    "lint": "tslint 'src/**/*.ts'  && npm run ts-coverage",
```
Comment (jimblanc) [inline packages/pushnotification/package.json:19]:
```suggestion
    "lint": "tslint 'src/**/*.ts' && npm run ts-coverage",
```

**Review by jimblanc — APPROVED** [review]

**Review by cshfang — APPROVED** [review]
Lgtm thanks!

**Review by svidgen — APPROVED** [review]

**Pull request merged**